vue-training/src/views/index/index.js

```diff
@@ -4,22 +4,51 @@ import { installServiceWorker } from '../../serviceWorkerInstaller'
 
 import TechService from '../../services/techsService.js'
 
+const MessageNoData = () => import('../../components/MessageNoData')
 const TechBox = () => import('../../components/TechBox')
-const TechTitle = () => import('../../components/TechTitle')
-
 
 import './index.pug'
 import './index.scss'
 
 const vm = new Vue({
   el: '#app',
   components: {
-    TechBox,
-    TechTitle
+    MessageNoData,
+    TechBox
   },
   data: {
+    filter: '',
     techs: []
   },
+  computed: {
+    filteredTechs() {
+      let newTechs = this.techs
+      const { filter } = this
+
+      if (filter) {
+        newTechs = newTechs.map(techGroup => {
+          const newTechsList = techGroup && techGroup.techsList.filter(tech => tech.title.toLowerCase().includes(filter.toLowerCase()))
+          const newTechGroup = {
+            ...techGroup,
+            techsList: newTechsList
+          }
+          return newTechGroup
+        })
+      }
+      return newTechs
+    },
+    countTechs() {
+      const countOfTechs = this.filteredTechs && this.filteredTechs.map(techGroup => techGroup && techGroup.techsList.length)
```

Comment:
Define a default value to `this.filteredTech` to avoid `&&` (same for techGroup)